File: run_install.go
package main

import (
	"fmt"
	"os"
	"os/exec"
	"path/filepath"
)

func main() {
	// Change to the directory
	dir := filepath.Join("C:", "Users", "aw789", "autonomous-github-agent")
	if err := os.Chdir(dir); err != nil {
		fmt.Printf("Error changing directory: %v\n", err)
		os.Exit(1)
	}

	fmt.Println("Running installation script...")
	fmt.Println("Directory:", dir)
	fmt.Println()

	// Run the Python installation script
	cmd := exec.Command("python", "install.py")
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr
	cmd.Stdin = os.Stdin

	if err := cmd.Run(); err != nil {
		fmt.Printf("Error running installation: %v\n", err)
		os.Exit(1)
	}

	fmt.Println("\nInstallation script completed.")
}

File: run_install.js
const { execSync } = require('child_process');
const path = require('path');
const process = require('process');

// Change to the autonomous-github-agent directory
const agentDir = 'C:\\Users\\aw789\\autonomous-github-agent';
process.chdir(agentDir);

console.log('========================================');
console.log('  Autonomous GitHub Agent Installation');
console.log('========================================\n');
console.log(`Working directory: ${agentDir}\n`);

try {
    // Run the Python installation script
    console.log('Executing install.py...\n');
    execSync('python install.py', {
        stdio: 'inherit',
        cwd: agentDir
    });

    console.log('\n✓ Installation completed successfully!');
} catch (error) {
    console.error('\n✗ Installation failed:', error.message);
    process.exit(1);
}
